src/compute/leaf.rs

```diff
@@ -25,19 +26,21 @@ pub(crate) fn compute(
 
     // Resolve node's preferred/min/max sizes (width/heights) against the available space (percentages resolve to pixel values)
     // For ContentSize mode, we pretend that the node has no size styles as these should be ignored.
-    let (node_size, node_min_size, node_max_size) = match sizing_mode {
+    let (node_size, node_min_size, node_max_size, aspect_ratio) = match sizing_mode {
         SizingMode::ContentSize => {
             let node_size = known_dimensions;
             let node_min_size = Size::NONE;
             let node_max_size = Size::NONE;
-            (node_size, node_min_size, node_max_size)
+            (node_size, node_min_size, node_max_size, None)
```

Comment:
So ContentSize will not apply aspect ratio but always keep to the size of the content? 

What will be the correct approach for a user to display an image in a node, but wanting to define what aspect ratio to use? Would it involve setting ContentSize on the image, and setting aspect ratio on the parent node instead?

Reply:
An image typically has an intrinsic aspect ratio (due to the ratio of the dimensions of the image file itself). But that would be out of scope for Taffy which doesn't know anything about images. The MeasureFunc for the image should apply this aspect ratio itself if it wants to. If the user wants to override this apsect ratio then they could set both a width (or height) and an aspect ratio on the leaf node and Taffy would apply this.

Medium term I suspect image nodes won't use Leaf layout with MeasureFuncs. Instead they will be their own custom layout modes (and responsible for managing their own styles outside of Taffy).

Reply:
Alright, thanks for the insight!